### ISSUE TITLE
[Bug] Segmentation Fault in Docker

### ISSUE THREAD

**Issue by ngysling** [BeamMP/BeamMP-Server#45]
Hi all, I'm trying to build a docker image for this binary, both for my convenience and to learn a little more about Docker. Here is the current version of my Dockerfile.

`
RUN \  
if [ -z ${BEAMMP_RELEASE+x} ]; then \                                                                                                          BEAMMP_RELEASE=$(curl -sX GET "https://api.github.com/repos/BeamMP/BeamMP-Server/releases/latest" \                                            | awk '/tag_name/{print $2;exit}' | sed 's/\x22//g' | sed 's/,*$//g'); fi \                                                              
&& \                                                                                                                                         
mkdir -p /app && \                                                                                                                                   
curl --location --progress-bar \                                                                                                                   
-o /app/BeamMP-Server https://github.com/BeamMP/BeamMP-Server/releases/download/${BEAMMP_RELEASE}/BeamMP-Server-linux 
`

This works, the binary downloads, running `file` on it returns as a Linux binary. Running it however, pops a segmentation fault error. Could this be an architecture/OS issue? The image is using Ubuntu.

**Comment by ngysling** [comment]
So I took an MD5 hash of the file, on my host I get 9d1ead73e678fa2f51a70a933b0bf017. In the container, I get 69bd950057da6381e4a79ab3903550a0. Can someone confirm if either of these are the correct hash?

**Comment by lionkor** [comment]
For convenience, please try to format the code you post. I did it here, and I took the liberty to change `/app` to `/tmp/app` to make it possible to test it on a non-docker system.
```sh
if [ -z ${BEAMMP_RELEASE+x} ]; then 
    BEAMMP_RELEASE=$(curl -sX GET "https://api.github.com/repos/BeamMP/BeamMP-Server/releases/latest" \
            | awk '/tag_name/{print $2;exit}' \
            | sed 's/\x22//g' \
            | sed 's/,*$//g');
fi
mkdir -p /tmp/app 
curl --location --progress-bar -o /tmp/app/BeamMP-Server https://github.com/BeamMP/BeamMP-Server/releases/download/${BEAMMP_RELEASE}/BeamMP-Server-linux 
```

This is, in essence, the same code you have and it works for me and downloads the correct executable (MD5 `69bd950057da6381e4a79ab3903550a0`), which runs fine. 

Your docker container also needs all dependencies listed in the README.md under "Prerequisites". Please verify that your docker container has all these libraries and that they are up to date. I think this might be the issue. If you keep getting segmentation faults, it would be great if you could run it in `gdb` instead and tell me where it crashes, if you know how to do that.

**Comment by ngysling** [comment]
Apologies for the late response, I never got an email. How did you format it properly? I did the code blocks in the editor, but anyway...

In regards to the dependencies, I have liblua5.3-dev libz-dev rapidjson-dev libssl-dev as well as git and curl for building the container. Am I correct in my assumption I don't need g++ and cmake if I'm using a binary? I will try it out anyway.     

Edit: I found a solution in using RouHims image that I somehow did not find in my search before I decided to build this.
